flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java

```diff
@@ -337,6 +346,264 @@ public void testSchedulingAllAtOnce() throws Exception {
+			CountDownLatchedAgnosticBinaryReceiver.setLatch(countDownLatch);
+
+			final SlotSharingGroup sharingGroup = new SlotSharingGroup(sender1.getID(), sender2.getID(), receiver.getID());
+			sender1.setSlotSharingGroup(sharingGroup);
+			sender2.setSlotSharingGroup(sharingGroup);
+			receiver.setSlotSharingGroup(sharingGroup);
+
+			final JobGraph jobGraph = new JobGraph("Bipartite job", sender1, sender2, receiver);
+
+			miniCluster.executeJobBlocking(jobGraph);
+		}
+	}
+
+	/**
+	 * A receiver that counts down the latch on running.
+	 */
+	public static class CountDownLatchedAgnosticBinaryReceiver extends AbstractInvokable {
+
+		private static CountDownLatch latch;
+
+		static void setLatch(CountDownLatch latch) {
+			CountDownLatchedAgnosticBinaryReceiver.latch = latch;
+		}
+
+		public CountDownLatchedAgnosticBinaryReceiver(Environment environment) {
+			super(environment);
+		}
+
+		@Override
+		public void invoke() throws Exception {
+			latch.countDown();
+
+			RecordReader<IntValue> reader1 = new RecordReader<>(
+				getEnvironment().getInputGate(0),
+				IntValue.class,
+				getEnvironment().getTaskManagerInfo().getTmpDirectories());
+
+			RecordReader<IntValue> reader2 = new RecordReader<>(
+				getEnvironment().getInputGate(1),
+				IntValue.class,
+				getEnvironment().getTaskManagerInfo().getTmpDirectories());
+
+			while (reader1.next() != null) { }
+			while (reader2.next() != null) { }
+		}
+	}
+
+	@Test
+	public void testSlotSharingForForwardJobWithFailedTaskManager() throws Exception {
+		final int parallelism = 20;
+		final int numOfTaskManagers = 2;
+
+		final Configuration configuration = getDefaultConfiguration();
+		configuration.setLong(JobManagerOptions.SLOT_REQUEST_TIMEOUT, 100L);
+
+		final MiniClusterConfiguration cfg = new MiniClusterConfiguration.Builder()
+			.setNumTaskManagers(numOfTaskManagers)
+			.setNumSlotsPerTaskManager(parallelism / numOfTaskManagers)
+			.setConfiguration(configuration)
+			.build();
+
+		try (final MiniCluster miniCluster = new MiniCluster(cfg)) {
+			miniCluster.start();
+
+			final JobVertex sender = new JobVertex("Sender");
+			sender.setInvokableClass(Sender.class);
+			sender.setParallelism(parallelism);
+
+			final JobVertex receiver = new JobVertex("BlockingReceiver");
+			receiver.setInvokableClass(BlockingReceiver.class);
+			receiver.setParallelism(parallelism);
+
+			receiver.connectNewDataSetAsInput(sender, DistributionPattern.POINTWISE,
+				ResultPartitionType.PIPELINED);
+
+			final SlotSharingGroup sharingGroup = new SlotSharingGroup();
+			sender.setSlotSharingGroup(sharingGroup);
+			receiver.setSlotSharingGroup(sharingGroup);
+
+			final JobGraph jobGraph = new JobGraph("Pointwise Job", sender, receiver);
+
+			final CompletableFuture<JobSubmissionResult> submissionFuture = miniCluster.submitJob(jobGraph);
+
+			final CompletableFuture<Void> failTaskManager = submissionFuture.thenRun(
+				() -> miniCluster.getTaskManagers()[0].shutDown());
```

Comment:
shutdown is not a blocking operation. You have to wait for the termination future instead, presumably using `thenCompose()`.

Reply:
you could also already call `get()` on the `submissionFuture`; if you aren't doing anything in-between chaining futures tends to just make things more complicated.
i.e.
```
miniCluster.submitJob(jobGraph).get();

TaskExecutor te = miniCluster.getTaskManagers()[0];
te.shutdown();
te.getTerminationFuture().get();

// check results
```